lib/relevantToExcursion.m

```diff
@@ -0,0 +1,23 @@
+%==========================================================================
+%This function takes in a list of objects and returns only those connected
+%to the specified excursionSet. It takes in 3 inputs.
+%
+%inputCell - a cell of input objects.
+%exNo - the number corresponding to the excursion set of interest.
+%labels - the corresponding label hashmap.
+%
+%Authors: Thomas Maullin, Camille Maumet.
+%==========================================================================
+function resultant = relevantToExcursion(inputCell, exNo, labels)
+    
+    counter = 1;
+    resultant = {};
+    %Work out which object belongs to which excursion set.
+    for(i = 1:length(inputCell))
```

Comment:
This loop is getting the keys (which are the nodes IDs) to access the map named 'labels'. It then uses the keys on the 'labels' map to obtain information about which excursion sets relate to which node.

I'm not sure how this could be done without having some loop calculating the keys... unless instead of the ID's, I made the key values the entire object... this seems like it would be a bit of a mess though